core/src/main/clojure/xtdb/vector/writer.clj

```diff
@@ -460,6 +460,14 @@
           (.writeNull absent-writer))
         (throw (UnsupportedOperationException. "populate-with-absents needs a nullable or union underneath!"))))))
 
+
+(defn- check-field-types [^FieldType expected-field-type ^FieldType given-field-type]
+  (or (and (= #xt.arrow/type :null (.getType expected-field-type))
+           (= #xt.arrow/type :null (.getType given-field-type)))
```

Comment:
This is a bit of weird one. I sometimes get `<FieldType :null null>` and `<FieldType :null not-null>` which then clash here.

Reply:
suspect that's a case of something indiscriminately calling `FieldType/notNullable` when it should probably be checking that the arrow-type is not `:null`